### PR TITLE
Modify ColorConverters to support two-way bindings

### PR DESCRIPTION
**Pull request by eduardoagr**
### Description of Change ###
 
Added a new HexToColor converter

 ### Linked Issues ###
 We need a way to convert the RGB color to a Maui color, that Labels Buttons, etc. understand. 

 - Fixes #

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

### CODE REVIEW

**Comment by brminnick** [comment]
Hi @eduardoagr! Is there a [New Feature Proposal](https://github.com/CommunityToolkit/Maui?tab=readme-ov-file#submitting-a-new-feature) opened for this feature?

All new features require a [New Feature Proposal](https://github.com/CommunityToolkit/Maui/issues/new?assignees=&labels=new%2Cproposal&projects=&template=open-a-new-feature-proposal.yml&title=%5BProposal%5D+) first, which will then be reviewed and approved by maintainers before a Pull Request can be merged.

**Comment by brminnick** [comment]
In your Proposal, please also explain how this is different than the existing [ColorToHexRgbaStringConverter](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/converters/color-to-hex-rgba-string-converter)

**Comment by brminnick** [comment]
Closing this PR until a New Feature Proposal has been approved

**Comment by bijington** [comment]
@brminnick there is a discussion where I agreed there is a bug in our converters, they are currently only one way so I suggested we would make them two way.

**Comment by brminnick** [comment]
Apologies. Re-opening.

**Comment by eduardoagr** [comment]
> Apologies. Re-opening.
> 
> Please update the PR Description. It is currently just the blank template.

how do I do that, this is my first contibrution 

also here is the discution 

https://github.com/CommunityToolkit/Maui/discussions/1496

**Comment by eduardoagr** [comment]
Aplogies for the mistake, I updated the template

I also want to update the documentation for converter, how I can do that?

I fout out hat in the documentation, there is only the guide of how to use this is XAML pages, but we can also make it globally

**Comment by eduardoagr** [comment]
Done

**Comment by bijington** [comment]
@eduardoagr sorry I didn't fully explain everything that was required because I thought the compiler would have provided the guidance. By changing the converters base class from `BaseConverterOneWay` to `BaseConverter` you will need to provide an implementation of the `ConvertBack` method. This will need to be the reverse of the `ConvertFrom` in each converter. This will then result in the converters providing a row way binding support and solving the issue that you originally came across.

As for the documentation we have a separate repository where you can submit a PR to. The repo is at: https://github.com/MicrosoftDocs/CommunityToolkit
I haven't checked the details directly but I think we just need to remove any mention of the converters being one-way in the documentation pages.

**Comment by eduardoagr** [comment]
Do you think we can schedule a team call sometime, to have more instrutin I also wanted t talk to you abount including a template   that already includes the MVVM Comunity toolkit and Maui Community toolkit

**Comment by bijington** [comment]
Yes I'd be happy to join a call and talk through the changes if that helps.

As for the template though it's not something we will implement however you can use the extension mentioned in your original discussion on the topic: https://github.com/CommunityToolkit/Maui/discussions/1505

**Comment by eduardoagr** [comment]
what is yur timezone, so I can program the tams metting

File: src/CommunityToolkit.Maui.UnitTests/Converters/BaseConverterTests.cs
Comment (pictos) [inline src/CommunityToolkit.Maui.UnitTests/Converters/BaseConverterTests.cs:9]: looks like your VS is overriding our editorconfig rules, can you fix it? I believe if you run the `dotnet format` command on your terminal it will fix those issues

Comment (eduardoagr) [reply src/CommunityToolkit.Maui.UnitTests/Converters/BaseConverterTests.cs:9]: You mean like forms, like indentation and stuff

Comment (eduardoagr) [reply src/CommunityToolkit.Maui.UnitTests/Converters/BaseConverterTests.cs:9]: Or run that command in the same folder

Comment (eduardoagr) [reply src/CommunityToolkit.Maui.UnitTests/Converters/BaseConverterTests.cs:9]: Well visual studio has a terminal

Comment (pictos) [reply src/CommunityToolkit.Maui.UnitTests/Converters/BaseConverterTests.cs:9]: like your `{` is in the same line as the method/class declaration

```cs
// your code is this way
public clas Bla {

    public void Do() {
    }
}

// we like this way
public clas Bla 
{

    public void Do() 
    {
    }
}

Comment (eduardoagr) [reply src/CommunityToolkit.Maui.UnitTests/Converters/BaseConverterTests.cs:9]: sorry about that

**Comment by VladislavAntonyuk** [comment]
The implementation looks not finished. You changed Rgb converter to BaseConverter, but Rgba remains BaseOneWayConverter.
also can we use Color.FromRgba(value) instead of Color.Parse?

**Comment by bijington** [comment]
I'm going to try and jump in this and get it finished off

**Comment by bijington** [comment]
> Color.FromRgba

`Color.FromRgba` doesn't support converting from the format: `RGBA(0,0,0,0)`

**Comment by bijington** [comment]
@eduardoagr apologies for the delay in getting this sorted. Would you mind please agreeing to the CLA?

**Comment by eduardoagr** [comment]
How do I sign

**Comment by eduardoagr** [comment]
@dotnet-policy-service agree

**Review by eduardoagr — COMMENTED** [review]
I just sign

**Comment by bijington** [comment]
> I just sign

That's great thank you

**Review by bijington — DISMISSED** [review]
Thanks for this @eduardoagr. I'll aim to update our docs shortly

**Review by bijington — APPROVED** [review]

**Pull request merged**
